test/gtest/query_mem.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <fstream>
+#include <filesystem>
+#include <optional>
+#include "file/file_utils.h"
+#include "nixl.h"
+#include "nixl_descriptors.h"
+
+class QueryMemTest : public ::testing::Test {
+protected:
+    void
+    SetUp() override {
+        // Create temporary test files
+        dir_path = "./files_for_query";
+        if (!(std::filesystem::exists(dir_path) && std::filesystem::is_directory(dir_path)))
+            std::filesystem::create_directory(dir_path);
```

Comment:
where is removed? also in other tests
and IMO better call the new dir ".test_files"